packages/firebase_ml_vision/lib/src/document_text_recognizer.dart

```diff
@@ -0,0 +1,179 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of firebase_ml_vision;
+
+/// Start or end of a component types detected by [DocumentTextRecognizedBreak].
+enum TextRecognizedBreakType {
+  unknown,
+  space,
+  sureSpace,
+  eolSureSpace,
+  hyphen,
+  lineBreak,
+}
+
+/// Detector for performing optical character recognition(OCR) on an input image.
+///
+/// In comparison to [TextRecognizer], it detects dense document text.
+///
+/// A document text recognizer is created via `cloudDocumentTextRecognizer()` in [FirebaseVision]:
+///
+/// ```dart
+/// final FirebaseVisionImage image =
+///     FirebaseVisionImage.fromFilePath('path/to/file');
+///
+/// final DocumentTextRecognizer documentTextRecognizer =
+///     FirebaseVision.instance.cloudDocumentTextRecognizer();
+///
+/// final List<VisionDocumentText> recognizedText =
+///     await documentTextRecognizer.processImage(image);
+/// ```
+class DocumentTextRecognizer {
+  DocumentTextRecognizer._({
+    @required int handle,
+  }) : _handle = handle;
+  final int _handle;
```

Comment:
nit: it's usually better to have final variables above the constructor

Reply:
https://dart.dev/guides/language/effective-dart/usage#do-initialize-fields-at-their-declaration-when-possible